libcaf_core/test/config_value.cpp

```diff
@@ -81,13 +86,739 @@ config_value cfg_lst(Ts&&... xs) {
+        CHECK_EQ(get_as<double>(x), 1.79769e+308);
+        CHECK_EQ(get_as<float>(x), sec::conversion_failed);
+      }
+    }
+  }
+  GIVEN("a config value x with value \"3e7\"") {
+    auto x = config_value{"3e7"s};
+    WHEN("using get_as with floating point types") {
+      THEN("CAF parses the string and converts the value") {
+        CHECK_EQ(get_as<long double>(x), 3e7);
+        CHECK_EQ(get_as<double>(x), 3e7);
+        CHECK_EQ(get_as<float>(x), 3e7f);
+      }
+    }
+  }
+  GIVEN("a config value x with value 123") {
+    auto x = config_value{123};
+    WHEN("using get_as with floating point types") {
+      THEN("CAF converts the value") {
+        CHECK_EQ(get_as<long double>(x), 123.0);
+        CHECK_EQ(get_as<double>(x), 123.0);
+        CHECK_EQ(get_as<float>(x), 123.f);
+      }
+    }
+  }
+  GIVEN("config_values of null, URI, boolean, list or dictionary") {
+    WHEN("using get_as with floating point types") {
+      THEN("conversion fails") {
+        CHECK_EQ(get_as<int64_t>(cv_null), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_true), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_false), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_uri), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_list), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_dict), sec::conversion_failed);
+      }
+    }
+  }
+}
+
+SCENARIO("get_as can convert config values to timespans") {
+  using namespace std::chrono_literals;
+  GIVEN("a config value with value 42s") {
+    auto x = config_value{timespan{42s}};
+    WHEN("using get_as with timespan") {
+      THEN("conversion succeeds") {
+        CHECK_EQ(get_as<timespan>(x), timespan{42s});
+        CHECK_EQ(get_as<std::string>(x), "42s");
+      }
+    }
+    WHEN("using get_as with type other than timespan or string") {
+      THEN("conversion fails") {
+        CHECK_EQ(get_as<int64_t>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<double>(x), sec::conversion_failed);
+        // CHECK_EQ(get_as<uri>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<config_value::list>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<config_value::dictionary>(x), sec::conversion_failed);
+      }
+    }
+  }
+}
+
+SCENARIO("get_as can convert config values to strings") {
+  using string = std::string;
+  GIVEN("any config value") {
+    WHEN("using get_as with string") {
+      THEN("CAF renders the value as string") {
+        CHECK_EQ(get_as<string>(cv_null), "null");
+        CHECK_EQ(get_as<string>(cv_true), "true");
+        CHECK_EQ(get_as<string>(cv_false), "false");
+        CHECK_EQ(get_as<string>(cv_empty_list), "[]");
+        CHECK_EQ(get_as<string>(cv_empty_dict), "{}");
+        CHECK_EQ(get_as<string>(config_value{42}), "42");
+        CHECK_EQ(get_as<string>(config_value{4.2}), "4.2");
+        CHECK_EQ(get_as<string>(config_value{timespan{4}}), "4ns");
```

Comment:
Since there is no other test case, just curious: Would `timespan{4s}` (assuming `using namespace std::chrono_literals;`) return a nanosecond resolution as well?

Reply:
No, that would be silly. Our timestamp has nanosecond resolution, so using this constructor gives you the default resolution.